js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
```

Comment:
The ergonomics of this API leave something to be desired. I think the validator would benefit greatly from a object-oriented refactoring:

```
var validator = new Validator(style);
validator.on('error', function(e) { ... });
validator.validate(); // entire style
validator.validateSource(source);
validator.validateLayer(layer);
validator.validatePaintProperty(key, value);
validator.validateLayoutProperty(key, value);
```

(Sorry I didn't realize this in the style-spec PR.)